src/helpers/load-script.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Holds a list of scripts that are still loading.
+ */
+let loadingQueue: string[] = [];
+
+interface ICallback {
+  id: string;
+  callback: () => unknown;
+}
```

Comment:
suggestion: I would recommend we define `interfaces` always BEFORE variables and other JS code, but after `import`s.